File: LeetCode/0201-0250/0226 Invert Binary Tree/main.js
function TreeNode(val) {
	this.val = val;
	this.left = this.right = null;
}

function invertTree(root) {
	if(root == null) {
		return root;
	}
	var tmp = root.left;
	root.left = root.right;
	root.right = tmp;
	invertTree(root.left);
	invertTree(root.right);
	return root
}

File: LeetCode/0201-0250/0226 Invert Binary Tree/main.cpp
struct TreeNode {
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
};

class Solution {
  public:
	struct TreeNode *invertTree(struct TreeNode *root) {
		if (root == nullptr) {
			return root;
		}
		invertTree(root->left);
		invertTree(root->right);
		TreeNode *tmp = root->left;
		root->left = root->right;
		root->right = tmp;
		return root;
	}
};
